[rubinnn] iP

rubinnn opened the pull request:
### Beeboo

>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

beeboo helps you to store and maintain a checklist.It's

- text-based

- easy to learn

- ~FAST~ _SUPER_ FAST to use

All you need to do is
1. Download the jar file from [here](https://github.com/rubinnn/ip/releases/tag/A-Jar)
2. Double click it
3. Add your tasks
4. Let it manage your task for **YOU**

Features:
- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Here is my Main class:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Level-0

Commit:
Level-1

Commit:
Level-2

Commit:
Level-3

Commit:
Level-4

Commit:
A-TextUiTesting

Commit:
Level-5

Commit:
Level-6

Commit:
Adjusted level 6

Commit:
Level-6 new adjustment

Commit:
Revert "Level-6 new adjustment"

This reverts commit 1f9f4cf36a3b6c4edf2016f813dbd20d6079cf40.

Commit:
Revert "Revert "Level-6 new adjustment""

This reverts commit 5ada9bad70eec1a61e9bed8204b226a6021b40b0.

Commit:
Branch is caught up

Commit:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Branch is caught up

Commit:
Level-7-inbranch

Commit:
Level-7 update

Commit:
Merge branch 'master' into branch-Level-7

* master:
  Revert "Revert "Level-6 new adjustment""
  Revert "Level-6 new adjustment"
  Level-6 new adjustment

Commit:
Level 7 update

Commit:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Level 7 update

Commit:
Level-7 Updated

Commit:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Level-7 Updated

Commit:
Level-8-branch

Commit:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Level-8-branch

Commit:
Create Command* classes:

Many different commands, more OOP to separate them into their own subclasses

Create more Commands under the superclass of Command with all similar attributes
of being commands to the chatbot

Create Ui, Storage, Parser, TaskList classes:

Extracted Ui, Storage, Parser and TaskList from main chatbot code,
to ensure better quality code with more OOP.

Ui, Storage, Parser and TaskList classes extracted and are called
on the main code with main code only running the chatbot.

Commit:
Merge branch 'A-MoreOOP'

* A-MoreOOP:
  Create Command* classes:

Commit:
Add packaging:

Add packaging to ensure readability of code

All files in the src/java folder are now in the beeboo package.
The files are then split into command, component as well as exceptions,
and put into their separate packages.

Commit:
Merge tag 'A-Packages'

* tag 'A-Packages':
  Add packaging:

Commit:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

Commit:
update A-Gradle

Commit:
Create Junit testing for methods in some classes

Ensure the code is working properly

Junit test created for the creation of tasks

Commit:
Merge tag 'A-JUnit'

* tag 'A-JUnit':
  Create Junit testing for methods in some classes

Commit:
A-jar

Commit:
Merge tag 'A-Jar'

* tag 'A-Jar':
  A-jar

Rububzz committed:
Add javadocs to public classes

Rububzz committed:
Merge tag 'A-JavaDoc'

* tag 'A-JavaDoc':
  Add javadocs to public classes

Rububzz committed:
Update formatting

Rububzz committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Update formatting

Rububzz committed:
Add FindCommand

To enable finding tasks in the tasklist

Rububzz committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add FindCommand

Rububzz committed:
Add A-Checkstyle

Rububzz committed:
Update style using CheckStyle

Rububzz committed:
Merge branch 'Branch-A-Checkstyle'

* Branch-A-Checkstyle:
  Update style using CheckStyle

Rububzz committed:
Add Gui to current project

Rububzz committed:
Update level-10

Rububzz committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Update level-10
  Add Gui to current project

zyng233 commented on src/main/java/beeboo/components/TimeConverter.java:
```diff
@@ -0,0 +1,35 @@
+            time = LocalTime.of(Integer.parseInt(arr[1].substring(0, 2)),
+                    Integer.parseInt(arr[1].substring(2)));
+        } else {
+            time = LocalTime.of(0, 0);
```
```suggestion
            time = LocalTime.of(0, 0);
```

zyng233 replied:
maybe can use a constant like time = LocalTime.MIDNIGHT;

zyng233 commented on src/main/java/beeboo/exception/InvalidCommandException.java:
```diff
@@ -0,0 +1,8 @@
+ * This exception is used to indicate that the command inputted does not exist.
+ */
+public class InvalidCommandException extends Exception {
+}
```
```suggestion
}
```

zyng233 replied:
can add a constructor to it

zyng233 commented on src/main/java/beeboo/command/AddCommand.java:
```diff
@@ -0,0 +1,72 @@
+            storage.saveItem(tasks);
+            return ui.addList(deadline, tasks.getSize());
+        default:
+            return "";
```
```suggestion
            return "";
```

zyng233 replied:
maybe can throw an exception such as IncvalidCommandException instead of return ""

zyng233 commented on src/main/java/beeboo/command/AddCommand.java:
```diff
@@ -0,0 +1,72 @@
+        case "d":
+            Deadlines deadline = Deadlines.createDeadline(command);
+            tasks.addList(deadline);
+            storage.saveItem(tasks);
```
```suggestion
            storage.saveItem(tasks);
```

zyng233 replied:
can factor the duplicated methods such as addList and saveList

zyng233 commented on src/main/java/beeboo/command/MarkCommand.java:
```diff
@@ -0,0 +1,62 @@
+            storage.saveItem(tasks);
+            return ui.unmarkDoneMessage(undone);
+        default:
+            return "";
```
```suggestion
            return "";
```

zyng233 replied:
can throw an excception as above

Quasant commented on src/main/java/beeboo/command/AddCommand.java:
```diff
@@ -0,0 +1,72 @@
+ */
+public class AddCommand extends Command {
+    private String type;
+    private String command;
```
For some of your subclasses of the Command class, you have a repeated attribute command. Perhaps consider making the attribute Protected, so that subclasses can access the inherited classes attributes.

Quasant commented on src/main/java/beeboo/components/Storage.java:
```diff
@@ -0,0 +1,137 @@
+    public ArrayList<Tasks> load() throws NoFileException {
+        File file = new File(filePath);
+        ArrayList<Tasks> list = new ArrayList<>();
+        if (!file.exists()) {
+            throw new NoFileException("");
+        } else {
```
Perhaps consider removing the else-condition; if the file doesn't exist and an exception is thrown, the code flow stops. This could help to reduce the level of indentations.

Quasant commented on src/main/java/beeboo/components/TimeConverter.java:
```diff
@@ -0,0 +1,35 @@
+     * @return a LocalDateTime object representing the provided date and time
+     * @throws DateTimeParseException if the date string is not in a valid format
+     */
+    public static LocalDateTime timeConverter(String date) throws DateTimeParseException {
```
Consider naming this method convertTime instead of timeConverter such that it is a verb instead of a noun.

Quasant reviewed (commented):
LGTM, mostly nitpicks about coding style.

matthewyip1511 commented on src/main/java/beeboo/Beeboo.java:
```diff
@@ -0,0 +1,93 @@
+    private TaskList tasks;
+    private Ui ui;
+
+    private static final String FILE_PATH = "./data/beeboo.txt";
```
Good use of `SCREAMING_SNAKE_CASE` here 👍

matthewyip1511 commented on src/main/java/beeboo/command/AddCommand.java:
```diff
@@ -0,0 +1,72 @@
+     */
+    public AddCommand(String type, String command) {
+        super(command);
+        this.command = command;
```
Agree with the above point, since you already used `super(command)` you won't need to use `this.command = command` again 🙂

matthewyip1511 commented on src/main/java/beeboo/command/AddCommand.java:
```diff
@@ -0,0 +1,72 @@
+            tasks.addList(todo);
+            storage.saveItem(tasks);
+            return ui.addList(todo, tasks.getSize());
+        case "d":
```
`t`, `d` and `e`are not very descriptive here, it might be better to type out in full as `todo`, `deadline` and `event`

matthewyip1511 commented on src/main/java/beeboo/task/Events.java:
```diff
@@ -0,0 +1,116 @@
+        String endDate = endDateCommand.substring(2).trim();
+        String[] endDates = endDate.split(" ");
+        LocalDateTime endDateTime = (endDates.length == 1)
+                ? TimeConverter.timeConverter(startDateTime.toLocalDate().toString() + " " + endDate)
```
this line is a bit long, you may want to consider adding a line break here 🙂

matthewyip1511 commented on src/main/java/beeboo/task/Deadlines.java:
```diff
@@ -0,0 +1,99 @@
+            throw new NoDescriptionException("No description");
+        }
+
+        String dateCommand = text.substring(descriptionEnd + 1).trim();
```
not too sure what `dateCommand` refers to here. If you meant to refer to the `deadline`, perhaps `deadlineDate` or deadline might work better 🙂

matthewyip1511 reviewed (commented):
Overall LGTM! Mainly just some issues with the naming of variables and some small coding standard violations, but otherwise good work 🙂

zyng233 reviewed (commented):
Overall it LGTM!

Rububzz committed:
Add Assertions

There are some functions that could possibly has certain assumptions that could
break the code if it is not tested. This helps to indicate any bugs to help find errors.

Rububzz committed:
Update code to Coding Standards

There were excess code that had no usage. To improve readability
the functions were deleted.

Rububzz committed:
Update code for code quality

There was more code that was redundant and was thus deleted

Rububzz committed:
Add Assertions

There was more code that needed assertions to ensure no breaking
of the code.Thus more assertions was added.

rubinnn committed:
Merge pull request #1 from rubinnn/A-Assertions

Add Assertions

Rububzz committed:
Merge branch 'master' into A-CodingQuality

* master:
  Add Assertions

rubinnn committed:
Merge pull request #2 from rubinnn/A-CodingQuality

Update code to follow code quality more closely

Rububzz committed:
Add CI

Used Github Actions to set up Continuous Integration (CI).

Rububzz committed:
Add duplicate detection

This prevents user from adding the same task more than once.
Ensures the user does not get confused when there is more than 1 task found
of the same name.

Rububzz committed:
Merge branch 'C-DetectDuplicates'

* C-DetectDuplicates:
  Add duplicate detection

Rububzz committed:
Update Code to ensure Code quality

There was some mistakes in coding quality, so code was updated
to ensure the coding was readable

Rububzz committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:
  Update Code to ensure Code quality

Rububzz committed:
Edit Test to follow coding guidelines

Coding style error was found and thus corrected

Rububzz committed:
Edit workflow

Workflow had other os that was not supported

Rububzz committed:
Update code to attempt CI

Deleted problematic codes that stopped CI from working

Rububzz committed:
Update code more to complie with CI

Problems with CI, thus trying to fix

Rububzz committed:
Add new extension

Added the UpdateCommand which allows user to update the dates of their events without
changing the description

Update Code quality

Updated some parts of the code according to PR reviews

Rububzz committed:
Merge branch 'C-Update'

* C-Update:
  Add new extension

Rububzz committed:
Update code for code quality

Update code with better code styling for better readability

rubinnn committed:
Update README.md

Update README to ensure that users are able to have a better understanding of how the chatbot works

Rububzz committed:
Add Ui.png

Add Ui.png to ensure that there is a good pictorial example
of how the chatbot would work and how it looks like

Rububzz committed:
Update README

Update README.md as there was some errors that occured

Rububzz committed:
Update code to ensure better SLAP

Update code to ensure that there methods are not too long
and added javadocs to ensure clarity of code

Rububzz committed:
Update README

Update README to ensure correct instructions of
Chatbot is shown.

rubinnn committed:
Update README.md

Update README to ensure that the 
instructions are accurate

Rububzz committed:
Merge branch 'master' of https://github.com/rubinnn/ip

* 'master' of https://github.com/rubinnn/ip:
  Update README.md

Rububzz committed:
Update CheckStyle

Update Checkstyle to ensure pass CI

Rububzz committed:
Update code to pass minimum requirement

Update code to ensure that there is no deep
nesting or too long methods

Rububzz committed:
Merge branch 'Adjusting-Code'

* Adjusting-Code:
  Update code to pass minimum requirement

Rububzz committed:
Update code to catch exceptions

Update code to ensure that non numeric
time output is caught

Rububzz committed:
Update code for style

Update code to fix style

Rububzz committed:
Update code for exceptions

Update code to ensure the code is working

Rububzz committed:
Update Readme and update function

Update Readme to have functions
and functions has been updated

Rububzz committed:
Update code to reduce length of methods

Update code to ensure methods are not too long

Rububzz committed:
Update code for style

Update code to ensure no styling errors

Rububzz committed:
Update ReadMe

Update ReadMe to allow hyperlinks to work

rubinnn committed:
Update README.md

Rububzz committed:
Update User guide

Update code to ensure user guide
matches readme

Rububzz committed:
Update README

Update README for the github pages to work

Rububzz committed:
Update README

Update README to the correct release for jar

Rububzz committed:
Update README

Update README with proper website
for releases
